test/milton/MiltonEvents.t.sol

```diff
@@ -0,0 +1,526 @@
+        uint256 closeTimestamp,
+        /// @notice account that liquidated the swap
+        address liquidator,
+        /// @notice asset amount after closing swap that has been transferred from Milton to the Buyer. Value represented in 18 decimals.
+        uint256 transferredToBuyer,
+        /// @notice asset amount after closing swap that has been transferred from Milton to the Liquidator. Value represented in 18 decimals.
+        uint256 transferredToLiquidator,
+        /// @notice incomeFeeValue value transferred to treasury
+        uint256 incomeFeeValue
+    );
+
+    event MiltonSpreadModelChanged(
+        address indexed changedBy, address indexed oldMiltonSpreadModel, address indexed newMiltonSpreadModel
+    );
+
+    function setUp() public {
+        _miltonSpreadModel = prepareMockSpreadModel(
+            TestConstants.ZERO, TestConstants.ZERO, TestConstants.ZERO_INT, TestConstants.ZERO_INT
+        );
+        _admin = address(this);
+        _userOne = _getUserAddress(1);
+        _userTwo = _getUserAddress(2);
+        _userThree = _getUserAddress(3);
+        _liquidityProvider = _getUserAddress(4);
+        _users = usersToArray(_admin, _userOne, _userTwo, _userThree, _liquidityProvider);
+    }
+
+    function testShouldEmitEventWhenOpenPayFixedSwap18Decimals() public {
+        // given
+        MockTestnetToken daiMockedToken = getTokenDai();
+        ItfIporOracle iporOracle = getIporOracleAsset(_userOne, address(daiMockedToken), 0);
+        IpToken ipTokenDai = getIpTokenDai(address(daiMockedToken));
+        MockCase0Stanley stanleyDai = getMockCase0Stanley(address(daiMockedToken));
+        MiltonStorage miltonStorageDai = getMiltonStorage();
+        ItfMiltonDai miltonDai = getItfMiltonDai(
+            address(daiMockedToken),
+            address(iporOracle),
+            address(miltonStorageDai),
+            address(_miltonSpreadModel),
+            address(stanleyDai)
+        );
+        ItfJosephDai josephDai = getItfJosephDai(
+            address(daiMockedToken),
+            address(ipTokenDai),
+            address(miltonDai),
+            address(miltonStorageDai),
+            address(stanleyDai)
+        );
+        prepareApproveForUsersDai(_users, daiMockedToken, address(josephDai), address(miltonDai));
+        prepareMilton(miltonDai, address(josephDai), address(stanleyDai));
+        prepareJoseph(josephDai);
+        prepareIpToken(ipTokenDai, address(josephDai));
+        // when
+        _miltonSpreadModel.setCalculateQuotePayFixed(TestConstants.PERCENTAGE_4_18DEC); // 4%
+        vm.prank(_userOne);
+        iporOracle.itfUpdateIndex(address(daiMockedToken), TestConstants.PERCENTAGE_3_18DEC, block.timestamp); // 3%, PERCENTAGE_3_18DEC
+        vm.prank(_liquidityProvider);
+        josephDai.itfProvideLiquidity(TestConstants.USD_28_000_18DEC, block.timestamp);
+        vm.prank(_userTwo);
+        vm.expectEmit(true, true, false, false);
```

Comment:
Why do you use `vm.expectEmit(true, true, false, false)` and not `vm.expectEmit(true, true, true, true)`